code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -105,7 +105,7 @@
 					playsound(E.loc, E.usesound, 50, 1)
 					to_chat(user, "<span class='notice'>Teleporting [src.name]...</span>")
 					E.teleporting = 1
-					if(!do_after(user, 50 * E.toolspeed, target = src))
+					if(!(do_after(user, 50 * E.toolspeed, target = src) && (E.rcell && (E.rcell.charge >= E.chargecost)) && Adjacent(user)))

```

Comment:
I'd probably check the charge and adjacency prior to attempting the do_after, perhaps before it makes the sound and shows the message, and have it instead show messages like "Not enough charge" or "You are too far from the object" or whatever.

Same for below.